better highlight

sandlerr opened the pull request:
@zendesk/samson 

fixes the problems in the previous PR:

- broken streaming output (horizontal scroll instead of vertical)
- annoying scroll event when clicking on a line of deploy log output

accidentally based this work on #806, I'll rebase if that isn't easily mergable

grosser commented:
after stopping a build it inserts newlines and then looks like this:

<img width="491" alt="screen shot 2016-04-11 at 8 07 46 am" src="https://cloud.githubusercontent.com/assets/11367/14431874/8392e132-ffbc-11e5-9878-cd8e6570fe42.png">

highlighting on click / on shift-click does not work ... seeing a js error "Uncaught TypeError: Cannot read property 'length' of undefined"

sandlerr commented:
so strange, it seems it has 3 states with different styles needed

1. during deploy - `block`
2. stopped deploy - `inline-block`
3. loaded a page that was already stopped - `inline` or `block`

I'll look more at it.

sandlerr commented:
fixed the JS and the separation. Turns out literal linebreaks are respected inside `<pre>` tags, and we insert them between lines... sometimes. I fixed it with CSS rather than resolving the root cause but it shouldn't create any more problems.

grosser commented on app/assets/stylesheets/deploys.scss:
```diff
@@ -135,3 +135,16 @@ td.status {
+
+#messages {
+  white-space: pre-wrap;
+  & > span {
```
`& >` .... just ` > ` ... you want all spans inside, right ?

grosser commented:
minor bugs: 
 - when continuing to shift-click it does not expand, but selects a new starting line
 - selecting line 3-10 breaks ... others work fine ... might be that there is a blank line ... not sure

grosser commented:
👍

sandlerr commented:
> when continuing to shift-click it does not expand, but selects a new starting line

That was intentional, but now I've clicked around I see that it should always expand the selection when shift-clicking.

> selecting line 3-10 breaks ... others work fine ... might be that there is a blank line ... not sure

yes, lines with no characters in them have a height of `0` :( I tried setting it to `18px` always but there is still a black gap so going to leave out that change.
